torch/_export/constraints.py

```diff
@@ -0,0 +1,36 @@
+from typing import Optional
+
+import torch._dynamo
+from torch._dynamo.comptime import comptime
+from torch.fx.experimental.symbolic_shapes import constrain_range
+
+
+# TODO: we want to hide this min/max stuff under some abstraction similar to
+# DynamicDim
+def add_inline_constraint(symbol, min: Optional[int] = None, max: Optional[int] = None):
+    """
+    Add min/max constraint on the intermediate symbol at tracing time
+    """
+
+    if torch._dynamo.is_compiling():
+        @comptime
+        def _(ctx):
+            node = ctx.get_local("symbol").as_proxy().node
+            min = ctx.get_local("min").as_python_constant()
```

Comment:
Shouldn't there be any checking going on here? Since `add_inline_constraint` is a user-facing API, they might make mistakes in passing the wrong types of inputs...

Reply:
I think they should be caught at runtime since the constrain_range function will be called while tracing, or at least be caught by a linter...